packages/core/src/util.ts

```diff
@@ -154,15 +158,23 @@ export async function getSourceForBacktrace(backtrace: BacktraceFrame[],
   return result
 }
 
-export function runBeforeNotifyHandlers(notice: Notice | null, handlers: BeforeNotifyHandler[]): boolean {
+export function runBeforeNotifyHandlers(notice: Notice | null, handlers: BeforeNotifyHandler[]): { results: ReturnType<BeforeNotifyHandler>[], result: boolean } {
```

Comment:
Hrm, it's confusing to me that `result` here will be `true` if the handlers return promises that eventually reject. However, I'm not sure of a better solution. 

One option could be something like an enum `status`... 

```javascript
function runBeforeNotifyHandlers(...) {
  const results = handlers.map(handler => handler(notice))

  if (results.find(result => result === false) {
    return { results, status: 'failed' }
  }

  if (results.find(result => result instanceof Promise) {
    return { results, status: 'pending' }
  }
  
  return { results, status: 'passed' }
}
```
Just an idea... the larger problem of mixing sync and async functions here is just thorny and I'm trying to think of ways it could be cleaner.

Reply:
If we decide not to try to return the status of the `beforeNotify` handlers to the user, perhaps this could be an async function and we could just wait for `Promise.allSettled()` before returning the result.

Reply:
Let's continue this discussion [here](https://github.com/honeybadger-io/honeybadger-js/pull/984#discussion_r1271868376).